lsp-mode.el

```diff
@@ -3872,8 +3882,9 @@ Need to handle the case when FILE indicates virtual buffer."
          ,@body)
      ,@body))
 
-(defun lsp--apply-text-document-edit (edit)
+(defun lsp--apply-text-document-edit (edit &optional operation)
```

Comment:
This require rather big change all the way until `lsp--apply-text-edits`.
Just my idea, would it be better to modify `edit` itself to include new value (for example `:_op?`) instead? And let `lsp--apply-text-edits` pull that value out and pass to the hook.
That can be nicely integrated with `lsp-protocol` too.

Reply:
I see your point, but op is not part of the protocol, not sure if this would trigger a whole new discussion

Reply:
Yeah, I think we've already adding new member to that structure started with `_`, with that it will not collide with protocol in any way.
The other approach is using dynamic binding like @yyoncho suggested.

Reply:
IMHO after @ericdallo fixed all these places current approach seems to be good enough. Although he touched a lot of places the change is low risk. @kiennq are you fine to proceed as it is?

Reply:
Okay, I'm fine with that.